Purchases/Support/BeginRefundRequestHelper.swift

```diff
@@ -61,23 +85,47 @@ class BeginRefundRequestHelper {
+     * Gets entitlement with the given `entitlementID` from customerInfo, or the active entitlement
+     * if no ID passed in.
      */
-    @MainActor
-    func beginRefundRequest(productID: String) async -> Result<RefundRequestStatus, Error> {
-        guard let windowScene = systemInfo.sharedUIApplication?.currentWindowScene else {
-            return .failure(ErrorUtils.storeProblemError(withMessage: "Failed to get UIWindowScene"))
-        }
+    func getEntitlement(maybeEntitlementID: String?) async throws -> EntitlementInfo {
+        let currentAppUserID = identityManager.currentAppUserID
+        return try await withCheckedThrowingContinuation { continuation in
+            customerInfoManager.customerInfo(appUserID: currentAppUserID) { maybeCustomerInfo, maybeError in
+                if let error = maybeError {
+                    let message = Strings.purchase.begin_refund_customer_info_error(
+                        entitlementID: maybeEntitlementID).description
+                    continuation.resume(
+                        throwing: ErrorUtils.beginRefundRequestError(withMessage: message, error: error))
+                    return
+                }
 
-        let transactionVerificationResult = await sk2Helper.verifyTransaction(productID: productID)
+                guard let customerInfo = maybeCustomerInfo else {
+                    let message = Strings.purchase.begin_refund_for_entitlement_nil_customer_info(
+                        entitlementID: maybeEntitlementID).description
+                    continuation.resume(throwing: ErrorUtils.beginRefundRequestError(withMessage: message))
+                    return
+                }
 
-        switch transactionVerificationResult {
-        case .failure(let verificationError):
-            return .failure(verificationError)
-        case .success(let transactionID):
-            return await sk2Helper.initiateRefundRequest(transactionID: transactionID,
-                                                         windowScene: windowScene)
+                if let entitlementID = maybeEntitlementID {
+                    guard let entitlement = customerInfo.entitlements[entitlementID] else {
+                        let message = Strings.purchase.begin_refund_no_entitlement_found(
+                            entitlementID: entitlementID).description
+                        continuation.resume(throwing: ErrorUtils.beginRefundRequestError(withMessage: message))
+                        return
+                    }
+                    continuation.resume(returning: entitlement)
+                    return
+                }
+
+                guard let activeEntitlement = customerInfo.entitlements.active.first?.value else {
+                    let message = Strings.purchase.begin_refund_no_active_entitlement.description
+                    continuation.resume(throwing: ErrorUtils.beginRefundRequestError(withMessage: message))
+                    return
```

Comment:
should we log all of these?

Reply:
i'm always confused where/when we should log. do we have a distinction on which kinds of errors to log? kinda makes me want to create a function that throws the error AND logs, or a function throwing the error that takes an optional log message or something. i know we talked about this once but i don't remember what the conclusion was..

Reply:
Yeah I think we need to improve how we do this in a bunch of places. It's very easy to end up not logging an error, or logging it twice because we log it when creating it and then when the parent receives it.

Reply:
i'll log for now, and create a ticket/find the existing one for improving error logging

Reply:
i created this in november: https://app.shortcut.com/revenuecat/story/10723/ios-refactor-logging-of-errors

Reply:
so it looks like `ErrorUtils`'s `logErrorIfNeeded` will log the `code.description`, the generic description for the type of error...it doesn't log the more specific message, though. that could be fixed by using sub-errors potentially..